core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -318,14 +319,14 @@ void toSerializableWithExtraRead(DistributedStorage storage)
     List<ParallelExecutorTask> tasks = new ArrayList<>();
 
     // Read set by scan is re-validated to check if there is no anti-dependency
-    for (Map.Entry<Scan, List<Key>> entry : scanSet.entrySet()) {
+    for (Map.Entry<ScanInterface, List<Key>> entry : scanSet.entrySet()) {
       tasks.add(
           () -> {
             Map<Key, TransactionResult> currentReadMap = new HashMap<>();
             Set<Key> validatedReadSet = new HashSet<>();
             Scanner scanner = null;
             try {
-              Scan scan = entry.getKey();
+              Scan scan = (Scan) entry.getKey();
```

Comment:
What will this code be like when RelationalScan is introduced?

Reply:
It's like the following, and `rebuildScan` will create another Scan operation using a builder pattern.

```java
              ScanInterface scan = entry.getKey();
              if (scan instanceof RelationalScan) {
                scanner =
                    storage.scan(
                        rebuildRelationalScan((RelationalScan) scan, getTableMetadata(scan)));
              } else {
                scanner = storage.scan(rebuildScan((Scan) scan, getTableMetadata(scan)));
              }
```

Reply:
Thanks! It would be great if we have 2 overloaded `rebuildScan(Scan)` and `rebuildScan(RelationalScan)` as we can simply write it like `scanner = storage.scan(rebuildScan(scan, getTableMetadata(scan)))`

Reply:
Even if we have 2 overloaded `rebuildScan`, I think we still need a cast and dispatch here since `scan` here is `ScanInterface`. Am I understanding correctly?